src/cpp/src/llm_pipeline_static.cpp

```diff
@@ -647,12 +660,287 @@ void stream_generated_tokens(std::shared_ptr<ov::genai::StreamerBase> streamer_p
     }
 }
 
+enum PipelineKind {
+    STATEFUL,
+    STATELESS
+};
+PipelineKind str_to_pipeline(const std::string& str) {
+    if (str == "STATEFUL") {
+        return PipelineKind::STATEFUL;
+    }
+    if (str == "STATELESS") {
+        return PipelineKind::STATELESS;
+    }
+    OPENVINO_THROW("Unsupported \"PIPELINE\" provided: " +
+                   str + ". Please select either \"STATEFUL\" or \"STATELESS\".");
+}
 } // anonymous namespace
 
 namespace ov {
 namespace genai {
+namespace static_llm {
 
-StaticLLMPipeline::StaticLLMPipeline(
+StatefulLLMPipeline::StatefulLLMPipeline(
+    const std::filesystem::path& models_path,
+    const ov::genai::Tokenizer& tokenizer,
+    const std::string&,
+    const ov::AnyMap& config
+) : LLMPipelineImplBase(tokenizer,
+                        utils::from_config_json_if_exists(models_path)) {
+
+    auto model = genai::utils::singleton_core().read_model(models_path / "openvino_model.xml", {}, config);
+    ModelConfigDesc model_desc = get_modeldesc_from_json(models_path / "config.json");
+    ov::AnyMap properties = config;
+
+    auto compiled = setupAndCompileModel(model, model_desc, properties);
+    m_request = compiled->create_infer_request();
+}
+
+
+StatefulLLMPipeline::StatefulLLMPipeline(
+    const std::shared_ptr<ov::Model>& model,
+    const ModelConfigDesc& model_desc,
+    const ov::genai::Tokenizer& tokenizer,
+    const std::string&,
+    const ov::AnyMap& properties,
+    const ov::genai::GenerationConfig& generation_config
+) : LLMPipelineImplBase(tokenizer, generation_config) {
+
+    bool use_blobs = false;
+    auto anyopt = get_option<bool>(properties, "USE_BLOBS");
+    if (anyopt.has_value()) {
+        use_blobs = *anyopt;
+    }
+    // Using model_str and weights_tensor with blobs is meaningless.
+    if (use_blobs) {
+        OPENVINO_THROW("Blobs cannot be used with model string and weights tensor");
```

Comment:
condition + throw = `OPENVINO_ASSERT(!use_blobs, "Blobs cannot be used with model string and weights tensor")`

the same in other places.

Reply:
Did this, Anatolii suggested `THROW` instead, @TolyaTalamanov?

Reply:
Sorry, I was wrong, I've used `OPENVINO_ASSERT(!use_blobs && "Blobs cannot be used with model string and weights tensor")` instead of `OPENVINO_ASSERT(!use_blobs, "Blobs cannot be used with model string and weights tensor")`. Fixed now

Reply:
I'd prefer having `Exception` rather than `Assert` in cases where the error is important for user...

Reply:
assert is the same as exception, but on C++ syntax level you write a single line instead of condition + throw

it's not C's assert which is working only in debug mode

Reply:
Ah ok, then it makes sense